[issue] panic: runtime error: invalid memory address or nil pointer dereference

**Describe the bug**
The command `./cloudlist -provider-config ../cloudlist.yaml` gives the error that I have attached in the screenshot topic. Tip: `./cloudlist -provider-config ../cloudlist.yaml -s compute,s3,dns,cloud-run,gke` is working fine, which means that the problem is in the `cloud-function`.

**Cloudlist version**
1.2.2 


**Screenshot of the error or bug**

```
./cloudlist -provider-config ../cloudlist.yaml

  _______             _____     __ 
 / ___/ /__  __ _____/ / (_)__ / /_
/ /__/ / _ \/ // / _  / / (_-</ __/
\___/_/\___/\_,_/\_,_/_/_/___/\__/ 

		projectdiscovery.io

[INF] Current cloudlist version 1.2.2 (latest)
[INF] Listing assets from provider: gcp services: compute,dns,cloud-function,cloud-run,s3,gke id: educorp-staging-e-production
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x10 pc=0x107a59b94]

goroutine 1 [running]:
github.com/projectdiscovery/cloudlist/pkg/providers/gcp.(*cloudFunctionsProvider).GetResource(0x14001d0f950, {0x140016194d0?, 0x10c08fb60?})
	github.com/projectdiscovery/cloudlist/pkg/providers/gcp/function.go:30 +0x124
github.com/projectdiscovery/cloudlist/pkg/providers/gcp.(*Provider).Resources(0x140016056e0, {0x109718cc0, 0x10c08fb60})
	github.com/projectdiscovery/cloudlist/pkg/providers/gcp/gcp.go:188 +0x384
github.com/projectdiscovery/cloudlist/internal/runner.(*Runner).Enumerate(0x140010bc320)
	github.com/projectdiscovery/cloudlist/internal/runner/runner.go:102 +0x674
main.main()
	./main.go:14 +0x74
```

--- a/pkg/providers/gcp/function.go
+++ b/pkg/providers/gcp/function.go
@@ -27,7 +27,7 @@ func (d *cloudFunctionsProvider) GetResource(ctx context.Context) (*schema.Resou
 		return nil, fmt.Errorf("could not get functions: %s", err)
 	}
 	for _, function := range functions {
-		if function.HttpsTrigger == nil {
+		if function == nil || function.HttpsTrigger == nil {
 			continue
 		}
 		funcUrl, err := url.Parse(function.HttpsTrigger.Url)

